sdk/java/pulumi/src/test/java/com/pulumi/core/UrnTest.java

```diff
@@ -1,60 +1,109 @@
 package com.pulumi.core;
 
-import com.pulumi.core.internal.Internal;
-import com.pulumi.deployment.MocksTest;
-import com.pulumi.deployment.internal.DeploymentTests;
-import com.pulumi.deployment.internal.TestOptions;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Test;
-
-import static com.pulumi.deployment.internal.DeploymentTests.cleanupDeploymentMocks;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.pulumi.core.OutputTests.waitForValue;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
 
 class UrnTest {
 
-    private static DeploymentTests.DeploymentMock mock;
-
-    @BeforeAll
-    public static void mockSetup() {
-        mock = DeploymentTests.DeploymentMockBuilder.builder()
-            .setMocks(new MocksTest.MyMocks())
-            .setOptions(new TestOptions(false))
-            .setMockGlobalInstance();
+    @SuppressWarnings({"unused"})
+    private static Stream<Arguments> testParseAndRender() {
+        return Stream.of(
```

Comment:
gathered the more interesting looking cases from greping other pulumi implementations